internal/controllers/machineset/machineset_controller.go

```diff
@@ -313,6 +313,10 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 
 	// Always updates status as machines come up or die.
 	if err := r.updateStatus(ctx, cluster, machineSet, filteredMachines); err != nil {
+		if errors.Is(err, remote.ErrClusterLocked) {
```

Comment:
If I see correctly what this achieves is that we requeue every minute instead of using the exponential backoff.

The idea with the exponential backoff was to ensure the queue doesn't get overloaded if there are a lot of Machines across various clusters and only a few clusters are unreachable.

Why do we need to requeue every minute?

Reply:
Ah is the problem that kerrors.NewAggregate doesn't preserve the ErrClusterLocked?

If that's the case/reason. Let's use Requeue: true here instead of RequeueAfter.

It would also be good to log the syncErr if it's != nil

Reply:
I think I would rewrite this section to
```go
	// Always updates status as machines come up or die.
	if err := r.updateStatus(ctx, cluster, machineSet, filteredMachines); err != nil {
		if syncErr != nil {
			return ctrl.Result{}, kerrors.NewAggregate([]error{syncErr, errors.Wrapf(err, "failed to update MachineSet's Status")})
		}
		if errors.Is(err, remote.ErrClusterLocked) {
			log.V(5).Info("Requeuing because another worker has the lock on the ClusterCacheTracker")
			return ctrl.Result{Requeue: true}, nil
		}
		return ctrl.Result{}, errors.Wrapf(err, "failed to update MachineSet's Status")
	}
```

If I understood it right this should still achieve what you wanted to do?

Reply:
Ah wait. Misinterpreted something. I'll rewrite the comments :)